File: Partner API V2/Java/src/OnescanSample/HMAC.java
package OnescanSample;

import java.io.UnsupportedEncodingException;
import java.net.HttpURLConnection;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;
import java.util.Formatter;

import javax.crypto.Mac;
import javax.crypto.SecretKey;
import javax.crypto.spec.SecretKeySpec;

public class HMAC {
	public static String HmacHeaderName = "x-onescan-signature";

	public static String Hash( String input, String keystring ) 
		throws NoSuchAlgorithmException, UnsupportedEncodingException, InvalidKeyException 
	{
		Mac mac = Mac.getInstance("HmacSHA1");

		//get the bytes of the hmac key and data string
		byte[] secretByte = keystring.getBytes("UTF-8");
	
		SecretKeySpec secret = new SecretKeySpec(secretByte, "HmacSHA1");

		mac.init(secret);
	
		byte[] dataBytes = input.getBytes("UTF-8");

		byte[] bytes = mac.doFinal(dataBytes);

		String form = "";
		for (int i = 0; i < bytes.length; i++) {
		    String str = Integer.toHexString(((int)bytes[i]) & 0xff);
		    if (str.length() == 1){
		        str = "0" + str;
		    }
	
		    form = form + str;
		}
		return form;		
	}

    public static void ValidateSignature(String input, String secret, String headerHmac) 
    		throws Exception {
        if (headerHmac == null || headerHmac.isEmpty())
            throw new Exception("HTTP Error: No signature found in the header");
        String calculatedHmac = HMAC.Hash(input, secret);
		if (!calculatedHmac.equals(headerHmac)) {
			throw new Exception("HTTP Error: Header signature is invalid");
		}
	}
}

File: Partner API V2/DotNet/OnescanSample/HMAC.cs
namespace OnescanSample {

    using System;
    using System.Configuration;
    using System.Net;
    using System.Security.Cryptography;
    using System.Text;
    using System.Web;

	/// <summary>
	/// Provides HMAC signing support for onescan messages.
	/// </summary>
	public class HMAC {

		/// <summary>
		/// The name of the header that is used to hold the message signature.
		/// </summary>
		public const string HmacHeaderName = "x-onescan-signature";

		/// <summary>
		/// Generating a hash for the input string, using the given key.
		/// </summary>
		/// <param name="input">The input string to hash</param>
		/// <param name="keystring">The key to use when hashing.</param>
		/// <returns>The hex encoded hash.</returns>
		public static string Hash( string input, string keystring ) {
			byte[] hash = null;
			byte[] key = Encoding.UTF8.GetBytes(keystring);
			byte[] byteArray = Encoding.UTF8.GetBytes(input);
	
			using (HMACSHA1 myhmacsha1 = new HMACSHA1(key)) {
				hash = myhmacsha1.ComputeHash(byteArray);
			}
			return string.Join("", Array.ConvertAll(hash, b => b.ToString("x2")));
		}

        public static void ValidateSignature(string input, System.Collections.Specialized.NameValueCollection headers = null) {
            if (headers == null)
            {
                headers = HttpContext.Current.Request.Headers;

            }
            string headerHmac = headers[HMAC.HmacHeaderName];
            if (string.IsNullOrEmpty(headerHmac))
                throw new Exception("HTTP Error: No signature found in the header");
            string calculatedHmac = HMAC.Hash(input, ConfigurationManager.AppSettings["OnescanSecret"]);
			if (calculatedHmac != headerHmac)
                throw new Exception("HTTP Error: Header signature is invalid");
        }
	}
}